src/Microsoft.Extensions.Caching.Abstractions/CacheEntryExtensions.cs

```diff
@@ -148,7 +136,6 @@ public static ICacheEntry SetOptions(this ICacheEntry entry, MemoryCacheEntryOpt
             entry.AbsoluteExpiration = options.AbsoluteExpiration;
             entry.AbsoluteExpirationRelativeToNow = options.AbsoluteExpirationRelativeToNow;
             entry.SlidingExpiration = options.SlidingExpiration;
-            entry.Priority = options.Priority;
```

Comment:
Not sure about this since EvictionMetadata may not make sense when all components share the same memory cache in DI. Would each component then have to know what to set the EvictionMetadata to? Also, what is the overhead of embedding the information in the value itself over having this field (would need to test this in a microbenchmark).